modules/platform/nuxeo-platform-video/src/test/java/org/nuxeo/ecm/platform/video/extension/TestVideoImporterAndListeners.java

```diff
@@ -482,4 +487,294 @@ public void testNoVideoConversionWhenDisabled() throws IOException {
+        // check video info
+        VideoDocument videoDocument = doc.getAdapter(VideoDocument.class);
+        Video video = videoDocument.getVideo();
+        assertEquals(320, video.getWidth());
+        assertEquals(200, video.getHeight());
+        assertEquals(0.05, video.getDuration(), 0.1);
+        // check previews
+        pictureViews = (List<Map<String, Serializable>>) doc.getPropertyValue("picture:views");
+        assertEquals(2, pictureViews.size());
+        assertEquals("Small", pictureViews.get(0).get("title"));
+        assertEquals("StaticPlayerView", pictureViews.get(1).get("title"));
+        // check storyboard
+        storyboard = (List<Map<String, Serializable>>) doc.getPropertyValue(STORYBOARD_PROPERTY);
+        assertEquals(2, storyboard.size());
+
+        transcodedVideos = (List<Map<String, Serializable>>) doc.getPropertyValue(TRANSCODED_VIDEOS_PROPERTY);
+        assertEquals(2, transcodedVideos.size());
+        assertEquals("MP4 480p", transcodedVideos.get(0).get("name"));
+        Blob transcodedVideoBlob = (Blob) transcodedVideos.get(0).get("content");
+        assertEquals("video/mp4", transcodedVideoBlob.getMimeType());
+        assertEquals("WebM 480p", transcodedVideos.get(1).get("name"));
+        transcodedVideoBlob = (Blob) transcodedVideos.get(1).get("content");
+        assertEquals("video/webm", transcodedVideoBlob.getMimeType());
+    }
+
+    // NXP-29966
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.video:test-video-conversions-enabled.xml")
+    @Deploy("org.nuxeo.ecm.platform.video:video-configuration-override.xml")
+    @SuppressWarnings("unchecked")
+    public void testComputeMissingOnVersion() throws IOException {
+        DocumentModel doc = session.createDocumentModel("/", "testVideoDoc", VIDEO_TYPE);
+        Blob blob = getBlobFromPath("test-data/sample.mpg", "video/mpeg");
+        doc.setPropertyValue("file:content", (Serializable) blob);
+        doc.putContextData(VideoChangedListener.DISABLE_VIDEO_CONVERSIONS_GENERATION_LISTENER, true);
+        // prefill video info and previews
+        Map<String, Serializable> videoInfoMap = new HashMap<>();
+        videoInfoMap.put(VideoInfo.WIDTH, 100);
+        videoInfoMap.put(VideoInfo.HEIGHT, 50);
+        videoInfoMap.put(VideoInfo.DURATION, 150);
+        videoInfoMap.put(VideoInfo.FORMAT, "foo");
+        doc.setPropertyValue(VideoConstants.INFO_PROPERTY, (Serializable) videoInfoMap);
+        List<Map<String, Serializable>> pictureviews = new ArrayList<>();
+        Map<String, Serializable> pictureView = new HashMap<>();
+        pictureView.put("title", "StaticPlayerView");
+        pictureView.put("content", (Serializable) Blobs.createBlob("dummy"));
+        pictureviews.add(pictureView);
+        doc.setPropertyValue("picture:views", (Serializable) pictureviews);
+        doc = session.createDocument(doc);
+        txFeature.nextTransaction();
+
+        // only video info and previews computed
```

Comment:
```suggestion
        // only video info and previews are set
```
as they actually haven't been computed by the listener